core/src/main/java/ai/timefold/solver/core/impl/domain/entity/descriptor/EntityDescriptorValidator.java

```diff
@@ -0,0 +1,151 @@
+package ai.timefold.solver.core.impl.domain.entity.descriptor;
+
+import java.lang.reflect.Member;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import ai.timefold.solver.core.api.domain.entity.PlanningEntity;
+import ai.timefold.solver.core.api.domain.entity.PlanningPin;
+import ai.timefold.solver.core.api.domain.entity.PlanningPinToIndex;
+import ai.timefold.solver.core.api.domain.variable.AnchorShadowVariable;
+import ai.timefold.solver.core.api.domain.variable.CascadingUpdateShadowVariable;
+import ai.timefold.solver.core.api.domain.variable.CustomShadowVariable;
+import ai.timefold.solver.core.api.domain.variable.IndexShadowVariable;
+import ai.timefold.solver.core.api.domain.variable.InverseRelationShadowVariable;
+import ai.timefold.solver.core.api.domain.variable.NextElementShadowVariable;
+import ai.timefold.solver.core.api.domain.variable.PiggybackShadowVariable;
+import ai.timefold.solver.core.api.domain.variable.PlanningListVariable;
+import ai.timefold.solver.core.api.domain.variable.PlanningVariable;
+import ai.timefold.solver.core.api.domain.variable.PreviousElementShadowVariable;
+import ai.timefold.solver.core.api.domain.variable.ShadowVariable;
+import ai.timefold.solver.core.config.util.ConfigUtils;
+
+public class EntityDescriptorValidator {
+
+    private static final Class[] PLANNING_ENTITY_ANNOTATION_CLASSES = {
```

Comment:
It only makes sense to extract the validation if it leads to _better_ code. If it creates duplication of some key logic (such as this array), it's not worth it and actually makes the code worse.

Reply:
The variables are slightly different. The one from `EntityDescriptorValidator` includes a few more types, and I did that to align with what I found in the Quarkus implementation.

Reply:
Any chance we could somehow merge the two arrays? Or rather make one "inherit" from the other?
I can already see us add a thing to one of them, and forget to add it to another - this is a bug waiting to happen.